libyul/backends/evm/StackLayoutGenerator.h

```diff
@@ -0,0 +1,87 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Stack layout generator for Yul to EVM code generation.
+ */
+
+#pragma once
+
+#include <libyul/backends/evm/ControlFlowGraph.h>
+
+#include <map>
+
+namespace solidity::yul
+{
+
+struct StackLayout
+{
+	struct BlockInfo
+	{
+		/// Complete stack layout that is required for entering a block.
```

Comment:
It is complete relative to function base, right?

Reply:
Yep